src/app/components/categories-details/categories-details.component.ts

```diff
@@ -86,23 +123,47 @@ export class CategoriesDetailsComponent {
       this.categoryId$,
       this.sort$,
       this.filters$,
+      this.storeSearch$,
     ]).pipe(
-      map(([freshProducts, categoryId, sortOption, filters]) => {
-        const freshProductsdetails = this._filterByCategoryIdAndMapToFPQuery(
-          freshProducts,
-          categoryId
-        ).filter((freshProduct) => {
-          let decider: boolean = true;
-          if (filters.priceFrom) {
-            decider = decider && freshProduct.price >= filters.priceFrom;
-          }
-          if (filters.priceTo) {
-            decider = decider && freshProduct.price <= filters.priceTo;
-          }
-          return decider;
-        });
-        return this._sort(freshProductsdetails, sortOption);
-      }),
+      map(
+        ([freshProducts, categoryId, sortOption, filters, storeSearchIds]) => {
+          const freshProductsdetails = this._filterByCategoryIdAndMapToFPQuery(
+            freshProducts,
+            categoryId
+          ).filter((freshProduct) => {
+            let decider: boolean = true;
+            if (filters.priceFrom) {
+              decider = decider && freshProduct.price >= filters.priceFrom;
+            }
+            if (filters.priceTo) {
+              decider = decider && freshProduct.price <= filters.priceTo;
+            }
+            if (filters.minRatingValue) {
+              decider =
+                decider && freshProduct.rating.value >= filters.minRatingValue;
+            }
+            if (
+              filters.storeIds.length > 0 &&
+              freshProduct.storeIds.length > 0
+            ) {
+              decider =
+                decider &&
+                filters.storeIds.some((id: string) =>
+                  freshProduct.storeIds.some((storeId) => storeId === id)
+                );
+            }
+            if (storeSearchIds?.length > 0) {
```

Comment:
Too many ifs. Try to do it without using so many ifs.

Reply:
I'll refractor it